frontend/src/components/pages/januscript/Documents.tsx

```diff
@@ -0,0 +1,36 @@
+import { useQuery } from "@apollo/client";
+import { Box, Typography } from "@mui/material";
+import React, { useEffect } from "react";
+
+import { ArchiveByTypesDocument } from "@/generated/graphql";
+
+import { DocumentList } from "./DocumentList";
+
+type Props = {
+  documentTypes: string[];
+  year: number | null;
+};
+
+export const Documents: React.FC<Props> = ({ documentTypes, year }) => {
+  const { refetch, loading, data, error } = useQuery(ArchiveByTypesDocument, {
+    variables: { documentTypes, year },
+    ssr: false,
+  });
+
+  useEffect(() => {
+    refetch({ documentTypes, year });
+  }, [documentTypes, year, refetch]);
```

Comment:
This is redundant, and I'll try to outline exactly why.

The way the Apollo Client, and conversely, the `useQuery` hook work is roughly something along these lines:


1. Your parent component renders, which has states such as `documentTypes` and `year`, i.e.
```tsx
[documentTypes, setDocumentTypes] = useState(...);
[year, setYear] = useState(...);
```
2. This component is rendered with the current values for the _stateful_ variables `year` and `documentTypes`, for example, this could be `year = 2023` and `documentTypes = ["regnskap"]` 
3. The `useQuery` hook is run with the the current value for `year` and `documentTypes`. The Apollo Client then uses these values in order to generate a _cache key_, which is uses to perform a lookup in the client-side cache. However, no such cache exists as this is the first time this query is run with those values.
4. Since the Apollo Client did not find anything in the client-side cache, it sends the request to our server, which returns list of documents. The Apollo Client then automatically adds the result to the _cache key_ we generated previously, so that any subsequent requests with the same values for `documentTypes` and `year` will hit the cache rather than send a new request to the server.
5. When the user then update the states in the parent component, e.g. `setYear(2022)`, the parent component will re-render, and all child components (importantly, including this component), will re-render. Now, this component will render with `documentTypes = ["regnskap"]` and `year = 2022` rather than `year = 2023`.
6. Same in 3., the `useQuery`-hook is run with the current values for `documentTypes` and `year`, and they're once again used to generate the cache key and we perform a cache-lookup. However, since the result of the previous render had `year = 2023` rather than `year = 2022`, we have no results for the newly generated cache key. As such, we repeat steps 4 and 5, sending a request to the server, storing the results in the new cache key, and rendering the component with the updated documents that match the user's preference.

**TL;DR:** If you update the state of one of the variables in a `useQuery`-hook, the query will be re-run and you'll get the updated results, basically refetching.

I've set up a simple [simple sandbox here to demonstrate this principle](https://stackblitz.com/edit/nextjs-nmozng?file=pages/index.js)

```suggestion
```